app/src/main/java/com/example/keniphhomemaintenance/dwellings/DwellingsViewModel.kt

```diff
@@ -1,29 +1,55 @@
 package com.example.keniphhomemaintenance.dwellings
 
+import android.content.Context
 import androidx.compose.runtime.MutableState
 import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.keniphhomemaintenance.CompanionObjects.Companion.PLACEHOLDER_ID
+import com.example.keniphhomemaintenance.R
+import com.example.keniphhomemaintenance.database.KeniphDatabase
+import com.example.keniphhomemaintenance.database.KeniphRepository
+import com.example.keniphhomemaintenance.maintenance_screen.MaintenanceItem
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 
-class DwellingsViewModel : ViewModel() {
+class DwellingsViewModel(context: Context) : ViewModel() {
 
-    private val placeholderDwelling1: Dwelling =
-        Dwelling(
-            "My home",
-            "420 Blazit Blvd., Rantoul, IL 61866"
-        )
-    private val placeholderDwelling2: Dwelling = Dwelling(
-        "Tenant complex",
-        "80085 Mayju Look., Rantoul, IL 61866"
-    )
-    val dwellingsList: MutableState<List<Dwelling>> = mutableStateOf(
-        listOf(
-            placeholderDwelling1,
-            placeholderDwelling2
-        )
-    )
+    private val repository: KeniphRepository
+    var dwellings: MutableState<List<Dwelling>> = mutableStateOf(listOf())
+    val loadedDwellingsPreviously: MutableState<Boolean> = mutableStateOf(false)
```

Comment:
loadedDwellingsPreviously prevents the LaunchedEffect block from executing again, upon the user navigating back to this screen, from the MaintenanceItemsScreen